### PR TITLE
add filter user service command 

### PR DESCRIPTION
**Pull request by borovHokage**

### CODE REVIEW

File: src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+
+    private async Task<List<OfficeFilteredData>> GetOfficeFilteredDataAsync(List<Guid> officesIds, List<string> errors)
+    {
+      if (officesIds == null || !officesIds.Any())
+      {
+        return null;
+      }
+
+      return (await RequestHandler.ProcessRequest<IFilterOfficesRequest, IFilterOfficesResponse>(
+       _rcGetOffices,
```
Comment (MariyaSin) [inline src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs]: add tab

File: src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+       IFilterOfficesRequest.CreateObj(officesIds),
+       errors, _logger)).Offices.ToList();
```
Comment (MariyaSin) [inline src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs]: move logger to next line

Comment (MariyaSin) [reply src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs]: do you need use toList?

File: src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs [hunk 3]
```diff
@@ -0,0 +1,411 @@
...
+    {
+      if (departmentsIds == null || !departmentsIds.Any())
+      {
+        return null;
+      }
+
+      return (await RequestHandler.ProcessRequest<IFilterDepartmentsRequest, IFilterDepartmentsResponse>(
+        _rcGetDepartments,
+        IFilterDepartmentsRequest.CreateObj(departmentsIds),
+        errors, _logger)).Departments.ToList();
```
Comment (MariyaSin) [inline src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs]: same

File: src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs [hunk 4]
```diff
@@ -0,0 +1,411 @@
...
+    {
+      if (positionsIds == null || !positionsIds.Any())
+      {
+        return null;
+      }
+
+      return (await RequestHandler.ProcessRequest<IFilterPositionsRequest, IFilterPositionsResponse>(
+        _rcGetPositions,
+        IFilterPositionsRequest.CreateObj(positionsIds),
+        errors, _logger)).Positions.ToList();
```
Comment (MariyaSin) [inline src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs]: same

File: src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs [hunk 5]
```diff
@@ -0,0 +1,411 @@
...
+    {
+      if (roleIds == null || !roleIds.Any())
+      {
+        return null;
+      }
+
+      return (await RequestHandler.ProcessRequest<IFilterRolesRequest, IFilterRolesResponse>(
+        _rcGetRoles,
+        IFilterRolesRequest.CreateObj(roleIds),
+        errors, _logger)).Roles.ToList();
```
Comment (MariyaSin) [inline src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs]: same

File: src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs [hunk 6]
```diff
@@ -0,0 +1,411 @@
...
+    {
+      if (filter.DepartmentsIds is null && 
+          filter.PositionsIds is null &&
+          filter.RightsIds is null && 
+          filter.OfficesIds is null)
+      {
+        return (await RequestHandler.ProcessRequest<IGetUsersDataRequest, IGetUsersDataResponse>(
+          _rcGetUsers,
+          IGetUsersDataRequest.CreateObj(new List<Guid>(), filter.SkipCount, filter.TakeCount),
+          errors, _logger)).UsersData.ToList();
```
Comment (MariyaSin) [inline src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs]: same

File: src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs [hunk 7]
```diff
@@ -0,0 +1,411 @@
...
+      }
+
+      return (await RequestHandler.ProcessRequest<IGetUsersDataRequest, IGetUsersDataResponse>(
+          _rcGetUsers,
+          IGetUsersDataRequest.CreateObj(usersIds),
+          errors, _logger)).UsersData.ToList();
```
Comment (MariyaSin) [inline src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs]: same

File: src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs [hunk 8]
```diff
@@ -0,0 +1,411 @@
...
+    }
+
+    private async Task<List<PositionData>> GetPositionsDataAsync(List<Guid> usersIds, List<string> errors)
+    {
+      return (await RequestHandler.ProcessRequest<IGetPositionsRequest, IGetPositionsResponse>(
+          _rcGetPositionsData,
+          IGetPositionsRequest.CreateObj(usersIds),
+          errors, _logger)).Positions.ToList();
```
Comment (MariyaSin) [inline src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs]: same

File: src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs [hunk 9]
```diff
@@ -0,0 +1,411 @@
...
+    }
+
+    private async Task<List<DepartmentData>> GetDepartmentsDataAsync(
+      List<Guid> usersIds,
+      List<string> errors)
+    {
+      return (await RequestHandler.ProcessRequest<IGetDepartmentsRequest, IGetDepartmentsResponse>(
+          _rcGetDepartmentsData,
+          IGetPositionsRequest.CreateObj(usersIds),
+          errors, _logger)).Departments.ToList();
```
Comment (MariyaSin) [inline src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs]: same

File: src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs [hunk 10]
```diff
@@ -0,0 +1,411 @@
...
+    {
+      if (usersImageIds == null || !usersImageIds.Any())
+      {
+        return null;
+      }
+
+      return (await RequestHandler.ProcessRequest<IGetImagesRequest, IGetImagesResponse>(
+        _rcGetImages,
+        IGetImagesRequest.CreateObj(usersImageIds, ImageSource.User),
+        errors, _logger)).ImagesData.ToList();
```
Comment (MariyaSin) [inline src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs]: same

File: src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs [hunk 11]
```diff
@@ -0,0 +1,411 @@
...
+
```
Comment (MariyaSin) [inline src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs]: remove empty line

File: src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs [hunk 12]
```diff
@@ -0,0 +1,411 @@
...
+        departmentsFilteredUsers?.SelectMany(x => x.UsersIds).ToList() ?? new List<Guid>(),
+        positionsFilteredData?.SelectMany(x => x.UsersIds).ToList() ?? new List<Guid>(),
+        rolesFilteredData?.SelectMany(x => x.UsersIds).ToList() ?? new List<Guid>());
+
+      //ToDo add emplementation SkipCount TakeCount
+      if ((filter.RightsIds is null &&
+           filter.OfficesIds is null &&
+           filter.DepartmentsIds is null &&
+           filter.PositionsIds is null) ||
+           filteredUsers is not null)
```
Comment (MariyaSin) [inline src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs]: fix tab

File: src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs [hunk 13]
```diff
@@ -0,0 +1,411 @@
...
+        }
+        else
+        {
+          positionInfo.AddRange(_positionInfoMapper.Map(positionsFilteredData));
+        }
+
+        if (filter.DepartmentsIds is null)
+        {
+          departmentData.AddRange(await GetDepartmentsDataAsync(
+                usersData.Select(u => u.Id).ToList(),
```
Comment (MariyaSin) [inline src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs]: fix tabs

File: src/FilterService.Models.Dto/Request/UserService/FilterUserService.cs [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+using Microsoft.AspNetCore.Mvc;
+
+namespace LT.DigitalOffice.FilterService.Models.Dto.Request.UserService
+{
+  public record FilterUserService : BaseFindFilter
+  {
+    [FromQuery(Name = "departmentsIds")]
+    public List<Guid> DepartmentsIds { get; set; }
+
+    [FromQuery(Name = "positionIds")]
```
Comment (MariyaSin) [inline src/FilterService.Models.Dto/Request/UserService/FilterUserService.cs]: positionS

File: src/FilterService/Startup.cs [hunk 15]
```diff
@@ -60,9 +71,9 @@ public Startup(IConfiguration configuration)
 
 	  _rabbitMqConfig = Configuration
 	    .GetSection(BaseRabbitMqConfig.SectionName)
-	    .Get<BaseRabbitMqConfig>();
+	    .Get<RabbitMqConfig>();
 
-	  Version = "1.0.0.0";
+	  Version = "1.0.1.0";
```
Comment (MariyaSin) [inline src/FilterService/Startup.cs]: set 1.0.0.0

File: src/FilterService/appsettings.json [hunk 16]
```diff
@@ -25,10 +25,33 @@
...
+
+    //PositionService
+    "FilterPositionsEndpoint": "PositionService_FilterPositions",
+    "GetPositionsEndpoint": "PositionService_GetPositions",
+
+    //DepartmentService
+    "FilterDepartmentsEndpoint": "DepartmentService_FilterDepartments",
+    "GetDepartmentsEndpoint": "DepartmentService_GetDepartments",
+
+    //RightService
```
Comment (MariyaSin) [inline src/FilterService/appsettings.json]: same as 30 string

File: src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs (hunk 1 above)
Comment (borovHokage) [reply src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs]: fixed

File: src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs (hunk 2 above)
Comment (borovHokage) [reply src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs]: fixed

File: src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs (hunk 11 above)
Comment (borovHokage) [reply src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs]: done

File: src/FilterService/Startup.cs (hunk 15 above)
Comment (borovHokage) [reply src/FilterService/Startup.cs]: fixed

File: src/FilterService/appsettings.json (hunk 16 above)
Comment (borovHokage) [reply src/FilterService/appsettings.json]: fixed

File: src/FilterService.Models.Dto/Request/UserService/FilterUserService.cs (hunk 14 above)
Comment (borovHokage) [reply src/FilterService.Models.Dto/Request/UserService/FilterUserService.cs]: done

File: src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs (hunk 12 above)
Comment (borovHokage) [reply src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs]: fixed

File: src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs (hunk 13 above)
Comment (borovHokage) [reply src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs]: fixed

File: src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs (hunk 12 above)
Comment (borovHokage) [reply src/FilterService.Business/Commands/User/UserServiceFilterCommand.cs]: done

File: src/FilterService/Controllers/UserServiceController.cs [hunk 17]
```diff
@@ -0,0 +1,23 @@
...
+using LT.DigitalOffice.FilterService.Models.Dto.Models;
+using LT.DigitalOffice.FilterService.Models.Dto.Request.UserService;
+using LT.DigitalOffice.Kernel.Responses;
+using Microsoft.AspNetCore.Mvc;
+
+namespace LT.DigitalOffice.FilterService.Controllers
+{
+  [Route("[controller]")]
+  [ApiController]
+  public class UserServiceController : ControllerBase
```
Comment (MariyaSin) [inline src/FilterService/Controllers/UserServiceController.cs]: UserController

File: src/FilterService/Controllers/UserServiceController.cs [hunk 18]
```diff
@@ -0,0 +1,23 @@
...
+  {
+    [HttpGet("filter")]
+    public async Task<FindResultResponse<UserInfo>> GetAsync(
+      [FromServices] IFilterUsersCommand command,
+      [FromQuery] FilterUserService filter)
```
Comment (MariyaSin) [inline src/FilterService/Controllers/UserServiceController.cs]: rename FilterUserService to UserFilter

File: src/FilterService.Models.Dto/Models/UserInfo.cs [hunk 19]
```diff
@@ -0,0 +1,18 @@
...
+
+namespace LT.DigitalOffice.FilterService.Models.Dto.Models
+{
+  public record UserInfo
+  {
+    public Guid Id { get; set; }
+    public string FirstName { get; set; }
+    public string LastName { get; set; }
+    public string MiddleName { get; set; }
+
```
Comment (MariyaSin) [inline src/FilterService.Models.Dto/Models/UserInfo.cs:12]: Add status from UserData

File: src/FilterService/Controllers/UserServiceController.cs (hunk 17 above)
Comment (borovHokage) [reply src/FilterService/Controllers/UserServiceController.cs]: fixed

File: src/FilterService.Models.Dto/Models/UserInfo.cs (hunk 19 above)
Comment (borovHokage) [reply src/FilterService.Models.Dto/Models/UserInfo.cs:12]: fixed

File: src/FilterService/Controllers/UserServiceController.cs (hunk 18 above)
Comment (borovHokage) [reply src/FilterService/Controllers/UserServiceController.cs]: done

File: src/FilterService.Business/Commands/User/FilterUsersCommand..cs [hunk 20]
```diff
@@ -0,0 +1,428 @@
...
+        departmentsFilteredUsers?.SelectMany(x => x.UsersIds).ToList() ?? new List<Guid>(),
+        positionsFilteredData?.SelectMany(x => x.UsersIds).ToList() ?? new List<Guid>(),
+        rolesFilteredData?.SelectMany(x => x.UsersIds).ToList() ?? new List<Guid>());
+
+      //ToDo add emplementation SkipCount TakeCount
+      if ((filter.RightsIds is null &&
+            filter.OfficesIds is null &&
+            filter.DepartmentsIds is null &&
+            filter.PositionsIds is null) ||
+            filteredUsers is not null)
```
Comment (ndoriomedov) [inline src/FilterService.Business/Commands/User/FilterUsersCommand..cs]: maybe better first check if filteredUsers is not null?

**Review by ndoriomedov — APPROVED** [review]

File: src/FilterService.Business/Commands/User/FilterUsersCommand..cs (hunk 20 above)
Comment (borovHokage) [reply src/FilterService.Business/Commands/User/FilterUsersCommand..cs]: ok

**Review by MariyaSin — APPROVED** [review]

**Pull request merged**